### ISSUE TITLE
Rechercher les usagers sans agent référent

### ISSUE THREAD

**Issue by yaf** [betagouv/rdv-solidarites.fr#2562]
## Actuellement

![Screenshot 2022-06-09 at 10-58-15 Vos usagers - RDV Solidarités](https://user-images.githubusercontent.com/42057/172808036-84ed86f6-2d27-4cc1-842e-95d08f073e5e.png)

Nous pouvons retrouver les usagers liés à un agent référent, mais rien ne nous permet de lister les usagers qui n'ont pas de référent.

## Objectif

Permettre une recherche d'usager n'ayant pas de référents.

**Comment by damienlethiec** [comment]
J'ai checké cette issue et il semble que ce soit déjà possible @yaf . Testé en local + le code ressemble à ça:
```
  def users
    return User.none if agent_id.blank? && search.blank?

    users = organisation.users
    users = users.merge(agent.users) if agent.present?
    users = users.search_by_text(search) if search.present?
    users
  end
```

**Comment by damienlethiec** [comment]
Sauf si j'ai mal compris et que le besoin est un index (une liste) de tous usagers sans référents. Dans ce cas en effet, ça manque

**Comment by damienlethiec** [comment]
En terme produit je propose l'option suivante (notamment car je n'aime pas bien l'UX qui pourrait être envisagée seulement avec select2 et sans onglet):

--> S'inspirer de la page des plages
![image](https://user-images.githubusercontent.com/8105430/190449930-3de42627-ef54-4bda-8ed4-0e33297306b7.png)

Quand on arrive sur la page, tous les utilisateurs s'affichent. J'ai accès à un champ de recherche et de sélection du référent (comme aujourd'hui). J'ai également accès à 3 onglets: tous les utilisateurs (sélectionné), suivis et non suivis. Quand je me rends sur suivis, je filtre les utilisateurs pour ne garder que ceux ayant un agent référent (et je garde mes résultats de recherche texte). Quand je me rends sur "non suivis", je filtre les utilisateurs pour que garder que ceux n'ayant pas d'agent référent (et je garde mes résultats de recherche texte) + le select2 de recherche de l'agent disparait.

Qu'en pensez-vous ?

**Comment by yaf** [comment]
Alors, dans le médico-social, il y a eu une demande liée à quelque chose « d'implicite » : la discrétion. Aussi, les agents ne souhaitent pas voir la liste des usagers.

Par contre, dans les autres domaines, cette discrétion n'est pas nécessaire (et c'est, il me semble, une demande qui a déjà été répertorié par @NesserineZarouri).

Nous pourrions vérifier auprès des référentes (avec Myriam) si cette discrétion est toujours nécessaire, ou bien, on pourrait avoir un `if` pour afficher la liste ou pas.

Ce point de base rend délicat la suite de ta proposition : si nous n'affichons pas de liste d'usager, difficile d'avoir des onglets ?

Peut-être faut-il ajouter simplement une entrée dans la liste des agents référents pour demander « sans référent ».


Les agents référents mériteraient d'être revue, mais ça n'enlève rien à cette demande-là.

**Comment by damienlethiec** [comment]
Yes, c'était l'autre option. Je ne crois pas que ce soit si simple avec select2 mais il doit y avoir moyen :)

**Comment by yaf** [comment]
ah oui, j'avais oublié ce fameux selec2... un peu pénible. Si tu vois d'autres options, c'est complétement ok aussi...

En y repensant, il y a un truc bizarre à afficher une liste d'usager sans référent, mais à ne pas afficher la liste principale. Je crois qu'il serait grand temps de reparler avec les référentes pour soit abandonner ce ticket, soit revenir en arrière sur la liste des usagers.

Merci pour tes questions. Il est mieux d'attendre avant de faire ce ticket. Je vais voir avec Myriam pour communiquer avec les référentes.

**Comment by damienlethiec** [comment]
Nickel :) Merci !

**Comment by damienlethiec** [comment]
Globalement, j'ai conscience que l'évolution produit proposée va au delà du sujet du ticket. Mais je trouve que ce genre de tickets sont justement l'occasion de requestionner des choix produits un peu plus larges qui seraient impactés. Pour éviter de se retrouver avec quelque chose de moins cohérent quelques tickets plus tard :)
